documentation/matlab-api.md

```diff
@@ -0,0 +1,385 @@
+   addRequired(p,'width',validScalarPosNum);
+   addOptional(p,'height',defaultHeight,validScalarPosNum);
+   addParameter(p,'units',defaultUnits,@isstring);
+   addParameter(p,'shape',defaultShape,...
+                 @(x) any(validatestring(x,expectedShapes)));
+   parse(p,width,varargin{:});
+   
+% (from MATLAB docs)
+```
+
+
+#### Pros
+- Very flexible
+- Well documented with potential online help
+- Standard MATLAB method
+- Explicit argument status
+- Keyword validation
+
+#### Cons
+- Known issue that an optional character input followed by a keyword-value pair will not give what you think
+- May change in future
+- Not extensible if doesn't meet requirements
+
+### `parse_arguments`
+PACE's `parse_arguments` supports positional, required and optional key-value pairs (`'name', value`) and flags (`-flagname`, `-noflagname`, unique substring)
+
+```matlab
+arglist=struct('fix_lattice',0,'fix_alatt',0,'fix_alatt_ratio',0,'fix_angdeg',0,...
+               'fix_orientation',0,'free_alatt',[1,1,1],'free_angdeg',[1,1,1],...
+               'bind_alatt',0);
+flags={'fix_lattice','fix_alatt','fix_alatt_ratio','fix_angdeg','fix_orientation'};
+[args,opt,present] = parse_arguments(varargin,arglist,flags);
+
+% (from refine_crystal)
+```
+
+#### Pros
+- Very flexible
+- Well documented
+- Under PACE control
+
+#### Cons
+- Bespoke, requires maintenance
+- Implicit state of arguments
+- Currently not explicitly tested
+
+### Raw `varargin` handler
+```matlab
+% Parse input arguments
+if nargin==7
+    expand_qe=false;    % set of distinct q points
+    ...
+    fwhh=varargin{7};
+elseif nargin==5
+    expand_qe=true;     % same q array for each energy in the energy array
+    ...
+    fwhh=varargin{5};
+elseif nargin==4
+    expand_qe=false;
+    ...
+    fwhh=varargin{4};
+else
+    error('Check number of input arguments')
+end
+
+(from disp2sqw)
+```
+
+#### Pros
+- Completely flexible handling of arguments specific to function
+
+#### Cons
+- Unreadable
+- Unmaintainable
+- Unintuitive
+- Breaks easily
+
+### Comparison
+
+| Type of Arg | `inputParser` | `parse_arguments` | Raw |
+|:-----------:|:-------------:|:-----------------:|:---:|
+| Positional  |      Y        |        Y          |  Y  |
+| Required    |      Y        |        Y          |  Y  |
+| Optional    |      Y        |        Y          |  Y  |
+| Key-value   |      Y        |        Y          |  Y  |
+| Flag        |      N        |        Y          |  Y  |
+| Default val |      Y        |        Y          |  Y  |
+| Validate value  |  Y        |        N          |  Y  |
+| Validate type |    Y        |        N          |  Y  |
+| Partial Matching |     Y        |        Y          |  Y  |
+
+## Options
+
+### Dash flags `-(no)flagname`
+
+Allow flags with dashes in MATLAB
+
+#### MATLAB
+
+Currently, using argument parser is flexible in the way it handles flags and allows the same flag (or its negation) to be specified in multiple ways.
```

Comment:
Is "argument parser" the Horace `parse_argument` function or the Matlab builtin `inputParser` class?